SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc

```diff
@@ -0,0 +1,241 @@
+#include "G4SystemOfUnits.hh"
+#include "G4HadProjectile.hh"
+#include "G4ElementVector.hh"
+#include "G4Track.hh"
+#include "G4Step.hh"
+#include "G4Element.hh"
+#include "G4ParticleChange.hh"
+#include "G4NucleiProperties.hh"
+#include "G4Nucleus.hh"
+
+#include "G4HadronicException.hh"
+#include "G4HadronicProcessStore.hh"
+#include "G4HadronicInteraction.hh"
+
+#include "G4ParticleDefinition.hh"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+G4SQInelasticProcess::G4SQInelasticProcess(double mass, const G4String& processName)
+    : G4HadronicProcess(processName, fHadronic) {
+  AddDataSet(new G4SQInelasticCrossSection(mass));
+  theParticle = G4SQ::SQ(mass);
+}
+
+G4SQInelasticProcess::~G4SQInelasticProcess() {}
+
+G4bool G4SQInelasticProcess::IsApplicable(const G4ParticleDefinition& aP) {
+  return theParticle->GetParticleType() == aP.GetParticleType();
+}
+
+G4VParticleChange* G4SQInelasticProcess::PostStepDoIt(const G4Track& aTrack, const G4Step& aStep) {
+  edm::LogInfo("G4SQInelasticProcess::PostStepDoIt")
+      << "Particle is going to interact at position" << aTrack.GetPosition() / cm
+      << " momentumdirection eta: " << aTrack.GetMomentumDirection().eta()
+      << " interacting in material : " << aTrack.GetMaterial() << std::endl;
+
+  // if primary is not Alive then do nothing
+  theTotalResult->Clear();
+  theTotalResult->Initialize(aTrack);
+  theTotalResult->ProposeWeight(aTrack.GetWeight());
+  if (aTrack.GetTrackStatus() != fAlive) {
+    edm::LogInfo("G4SQInelasticProcess::PostStepDoIt") << "No interaction because primary is not alive" << std::endl;
+    return theTotalResult;
+  }
+
+  edm::LogInfo("G4SQInelasticProcess::PostStepDoIt") << "Start a possible interaction?" << std::endl;
+
+  if (aTrack.GetPosition().rho() / centimeter < 1) {
+    edm::LogInfo("G4SQInelasticProcess::PostStepDoIt")
+        << "FYI: the rho of the track is < 1cm and it's still going to interact..." << std::endl;
+  }
+
+  // Find cross section at end of step and check if <= 0
+  //
+  G4DynamicParticle* aParticle = const_cast<G4DynamicParticle*>(aTrack.GetDynamicParticle());
+
+  G4Material* aMaterial = aTrack.GetMaterial();
+
+  const G4Element* anElement = 0;
+  try {
+    anElement = theCrossSectionDataStore->SampleZandA(aParticle, aMaterial, targetNucleus);
+  } catch (G4HadronicException& aR) {
+    G4ExceptionDescription ed;
+    aR.Report(ed);
+    DumpState(aTrack, "SampleZandA", ed);
+    ed << " PostStepDoIt failed on element selection" << G4endl;
+    G4Exception("G4SQInelasticProcess::PostStepDoIt", "had003", FatalException, ed);
+  }
+
+  // check only for charged particles
+  if (aParticle->GetDefinition()->GetPDGCharge() != 0.0) {
+    if (GetElementCrossSection(aParticle, anElement, aMaterial) <= 0.0) {
+      // No interaction
+      return theTotalResult;
+    }
+  }
+
+  // Next check for illegal track status
+  //
+  if (aTrack.GetTrackStatus() != fAlive && aTrack.GetTrackStatus() != fSuspend) {
```

Comment:
there is no sense in this check also.

Reply:
Ok, anyway whole method removed as per your later comment.